public/friendlycode/js/fc/load-project-files.js

```diff
@@ -0,0 +1,168 @@
+define([], function() {
+  "use strict";
+
+  var Path = Bramble.Filer.Path;
+  var FilerBuffer = Bramble.Filer.Buffer;
+  var self = {};
+
+  // Taken from http://stackoverflow.com/questions/6965107/converting-between-strings-and-arraybuffers
+  function convertToBuffer(string) {
+    var buf = new ArrayBuffer(string.length);
+    var bufView = new Uint8Array(buf);
+    for(var i = 0, strLen = string.length; i < strLen; i++) {
+      bufView[i] = string.charCodeAt(i);
+    }
+    return buf;
+  }
+
+  self.load = function(project, defaultTemplate, callback) {
+    var fs = Bramble.getFileSystem();
+    var shell = new fs.Shell();
+    var projectFilesUrl = window.location.protocol + "//" + window.location.host + "/initializeProject";
+    var request = new XMLHttpRequest();
```

Comment:
It might be nice to use native XHR, so we can move this whole thing into a worker thread in the future.